lib/internal/fs/streams.js

```diff
@@ -386,22 +386,50 @@ WriteStream.prototype.open = openWriteFs;
 
 WriteStream.prototype._construct = _construct;
 
+function writeAll(data, pos, cb) {
+  const len = data.length;
+  this[kFs].write(this.fd, data, 0, len, pos, (er, bytesWritten) => {
+    if (this.destroyed || er) {
+      return cb(er);
```

Comment:
This is _not_ correct, you might get a `'EAGAIN'` error and you'd need to retry with that too.